api/models/vpp/srv6/srv6.proto

```diff
@@ -0,0 +1,123 @@
+    }
+
+    /* End function behavior of endpoint with specific IPv6 table lookup*/
+    message EndT {
+        bool psp = 1;                    /* use PSP (penultimate segment POP of the SRH) or by default use USP (Ultimate Segment Pop of the SRH) */
+    }
+
+    /* End function nehavior of endpoint with decapsulation and Layer-2 cross-connect (or DX2 with egress VLAN rewrite when VLAN notzero - not supported this variant yet)*/
+    message EndDX2 {
+        uint32 vlan_tag = 1;             /* Outgoing VLAN tag */
+        string outgoing_interface = 2;   /* name of cross-connected outgoing interface */
+        string next_hop = 3;             /* next hop address for cross-connected link */
+    }
+
+    /* End function behavior of endpoint with decapsulation and IPv4 cross-connect*/
+    message EndDX4 {
+        string outgoing_interface = 1;   /* name of cross-connected outgoing interface */
+        string next_hop = 2;             /* next hop address for cross-connected link */
+    }
+
+    /* End function behavior of endpoint with decapsulation and IPv6 cross-connect*/
+    message EndDX6 {
+        string outgoing_interface = 1;   /* name of cross-connected outgoing interface */
+        string next_hop = 2;             /* next hop address for cross-connected link */
+    }
+
+    /* End function behavior of endpoint with decapsulation and specific IPv4 table lookup*/
+    message EndDT4 {
+        // TODO model this end function
+    }
+
+    /* End function behavior of endpoint with decapsulation and specific IPv6 table lookup*/
+    message EndDT6 {
+        // TODO model this end function
+    }
+
+    /* End function behavior of dynamic segment routing proxy endpoint */
+    message EndAD {
+        string service_address = 1;      /* IPv6/IPv4 address of SR-unaware service (address type depends whether service is IPv4 or IPv6 service) */
+        string outgoing_interface = 2;   /* name of interface on segment routing proxy side sending data to segment routing unaware service */
+        string incoming_interface = 3;   /* name of interface on segment routing proxy side receiving data from segment routing unaware service */
+    }
+}
+
+
+/* Model for SRv6 policy (policy without at least one policy segment is only cached in ligato and not written to VPP)*/
+message Policy {
+    string bsid = 1;                     /* binding SID (IPv6 Address) */
+    uint32 fib_table_id = 2;             /* ID of FIB table where to install Policy entry */
+    bool srh_encapsulation = 3;          /* are SR headers handled by encapsulation? (no means insertion of SR headers) */
+    bool spray_behaviour = 4;            /* spray(multicast) to all policy segments? (no means to use PolicySegment.weight to loadbalance traffic)*/
+}
+
+/* Model for segment list of SRv6 policy */
+message PolicySegmentList {
+    string policy_bsid = 1;             /* reference to policy where this policy segment belongs */
```

Comment:
Model PolicySegmentList doesn't mention any name, yet the key is composed of it -- apparently for this reason ModelPolicySegmentList couldn't be used to generate and parse keys (hence the method ParsePolicySegmentList, the sort of which we want to get rid of). Would it be possible to add the logical name here and register model with template combining the name with the policy bsid?

Reply:
The logical name is there only because key must be unique (name + policy_bsid = unique). The proper unique combination (from proto model fields only) would be from all fields (policy_bsid + wight + segments), but that could be possible really long key (segments field is list and it's length is not limited by segment routing standards). Therefore i prefer the logical name, but i don't want to pollute the model with it. I didn't know that you wanted to get rid of all custom parse methods, but now it seems to me like a good trade-off (1 custom parse method vs polluted model).

Reply:
It has to be possible to generate key using only the content of the value - i.e. the Name must be included - otherwise GRPC/local client will not work correctly. It is because the values to configure are given to the client without keys (unlike when KVDB is used as NB interface) and the client uses the models to build the appropriate key for every value, e.g.: https://github.com/ligato/vpp-agent/blob/master/client/local_client.go#L53

Reply:
Ok, i have to change it so that key can be generated from value. For PolicySegmentList it means to add name to model (merge of policy and policysegmentlist as suggested https://github.com/ligato/vpp-agent/pull/1174#issuecomment-464697745 would also fix this). I also have the same problem in steering (https://github.com/ligato/vpp-agent/pull/1174/files#diff-60062a829b92c1b0f46b0d677ae8764bR103) where i use in key name that is not in value. I will add the name into steering model too.

Reply:
Name moved into models in https://github.com/ligato/vpp-agent/pull/1174/commits/edd20eea90f66870a18d336731c8de81b2365019 . 
However this doesn't removes ParsePolicySegmentList mentioned in first comment in this thread, because there is no standard way how to extract policyBSID from PolicySegmentList key (my first usage of that method) and also there is no proper name validation (my second usage) (https://github.com/ligato/vpp-agent/blob/42cf54a33060b64cd4b8ff3c7418a4e1798bc9d6/pkg/models/spec.go#L87)